### PR TITLE
Refactored to separate public APIs into top-level /api folder; fixes …

### PR DESCRIPTION
**Pull request by tomikazi**
…issue #824

### CODE REVIEW

**Comment by tomikazi** [comment]
This represents a bit of a departure from the standard Go project layout however. I’d like to see Sean’s, Andrea’s and Adib’s view on this. I’m wondering whether we ought to put it under /pkg/api instead to stick with the convention and still achieve the desired goal. It would be Go-specific, but it really already is the way it is now.

File: pkg/cli/networkchanges_test.go [hunk 1]
```diff
@@ -18,10 +18,10 @@ package cli
 import (
 	"bytes"
 	"fmt"
-	"github.com/onosproject/onos-config/pkg/northbound/diags"
-	"github.com/onosproject/onos-config/pkg/types/change"
-	"github.com/onosproject/onos-config/pkg/types/change/device"
-	"github.com/onosproject/onos-config/pkg/types/change/network"
+	"github.com/onosproject/onos-config/api/diags"
+	"github.com/onosproject/onos-config/api/types/change"
+	"github.com/onosproject/onos-config/api/types/change/device"
```
Comment (SeanCondon) [inline pkg/cli/networkchanges_test.go]: should use consistent prefix throughout for this package - called `devicechange` else where

File: pkg/controller/change/device/computedelta_test.go [hunk 2]
```diff
@@ -16,10 +16,10 @@ package device
 
 import (
 	"fmt"
+	"github.com/onosproject/onos-config/api/types"
+	"github.com/onosproject/onos-config/api/types/change"
+	devicechangetypes "github.com/onosproject/onos-config/api/types/change/device"
```
Comment (SeanCondon) [inline pkg/controller/change/device/computedelta_test.go]: This is why `github.com/onosproject/onos-config/api/types/change/device` was given the alias `devicechangetypes` consistently through out the code. `devicechanges` had been used for the package `github.com/onosproject/onos-config/pkg/types/change/device`

I don't care what alias is used for either as long as the usage is consistent throughout the project. It includes code readability and consistency

**Comment by Andrea-Campanella** [comment]
@tomikazi I like the approach of having the APIs in the too level `/api` folder, it makes it easier for somebody to go and check them out. Kubernetes takes it one step further with https://github.com/kubernetes/api repo, but I prefer the separation ATM. 
I also agree with both @SeanCondon comments, you need to be consistent with the aliases across the whole repo or it make a big mess of understanding exactly what we are pointing at. 
refactoring it in small chunks also makes it easier to spot if there is an issue. I'd split this one up in multiple PRs.

**Comment by tomikazi** [comment]
Issuing a new PR with resolved conflicts.

**Pull request closed without merging**